submission/loglist_manager.go

```diff
@@ -21,8 +21,25 @@ import (
 	"time"
 
 	"github.com/google/certificate-transparency-go/schedule"
+	"github.com/google/trillian/monitoring"
 )
 
+var (
+	logRefOnce          sync.Once
+	logListSinceRefresh monitoring.Gauge // updates every minute
+)
+
+func incTimeSinceRefresh(_ context.Context) {
+	logListSinceRefresh.Inc()
```

Comment:
Inc() seems an odd thing to do here.  If the metric is gauge, and described as 'time since ...'
then I'd expect you to set it to the minute duration.  Inc() seems more relevant for a counter metric.

But I see you're using a minute interval timer to count each minute as it goes by.
Do you do this because it's hard to obtain the timestamp that the refresh started, and the timestamp the refresh finished, and to set the gauge to the difference (in minutes)?  Or because you want a running total as things are in progress?

Reply:
hm, this metric should track not the length of refresh process, but rather time since last refresh happened. Meaning it should be dynamic.

Reply:
Per offline discussion switched to Counter

Reply:
discussed in person: try using a Counter instead of a Gauge, as you don't -really- want to set lots of arbitrary values.

If you can Inc() most of the time and occasionally Reset() the Counter to zero, then that would be ideal.

But if the Counter doesn't let you Reset() then you might need to stick with gauge.

in my experience Counter metrics are really ideally suited to things that you want to graph the rate of change of, and where you tend not to care about absolute counter values.

Reply:
If you just record the time the last refresh happened in the metric, Monarch can calculate how long ago that was and give you a nice smooth graph.

Reply:
That's a good point, this approach will always lead to a staircase graph.
Is this a good example of not trying to do maths that the monitoring system can do for you?

Reply:
That's a valid point. Thanks, Rob. Updated the metric and the code.